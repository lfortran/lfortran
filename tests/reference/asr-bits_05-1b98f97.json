{
    "basename": "asr-bits_05-1b98f97",
    "cmd": "lfortran --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/bits_05.f90",
    "infile_hash": "9a68b4d514588f51399782270c3a63efb0fe5d3fce1c4617547842cc",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-bits_05-1b98f97.stdout",
    "stdout_hash": "2657f298e62697dc98be06a75ee1e47119dbc445a02d864d1f51fc32",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}