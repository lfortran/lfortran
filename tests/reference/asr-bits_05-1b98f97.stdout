(TranslationUnit
    (SymbolTable
        1
        {
            bits_05:
                (Program
                    (SymbolTable
                        2
                        {
                            from:
                                (Variable
                                    2
                                    from
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            from8:
                                (Variable
                                    2
                                    from8
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 8)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            res:
                                (Variable
                                    2
                                    res
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            res8:
                                (Variable
                                    2
                                    res8
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 8)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    bits_05
                    []
                    [(Assignment
                        (Var 2 from)
                        (IntegerConstant 10 (Integer 4))
                        ()
                    )
                    (Assignment
                        (Var 2 from8)
                        (IntegerConstant 10 (Integer 8))
                        ()
                    )
                    (Assignment
                        (Var 2 res)
                        (IntrinsicElementalFunction
                            Ibits
                            [(Var 2 from)
                            (IntegerConstant 2 (Integer 4))
                            (IntegerConstant 2 (Integer 4))]
                            0
                            (Integer 4)
                            ()
                        )
                        ()
                    )
                    (If
                        (IntegerCompare
                            (Var 2 res)
                            NotEq
                            (IntegerConstant 2 (Integer 4))
                            (Logical 4)
                            ()
                        )
                        [(ErrorStop
                            ()
                        )]
                        []
                    )
                    (Assignment
                        (Var 2 res8)
                        (IntrinsicElementalFunction
                            Ibits
                            [(Var 2 from8)
                            (IntegerConstant 2 (Integer 4))
                            (IntegerConstant 2 (Integer 4))]
                            0
                            (Integer 8)
                            ()
                        )
                        ()
                    )
                    (If
                        (IntegerCompare
                            (Var 2 res8)
                            NotEq
                            (IntegerConstant 2 (Integer 8))
                            (Logical 4)
                            ()
                        )
                        [(ErrorStop
                            ()
                        )]
                        []
                    )
                    (Assignment
                        (Var 2 res)
                        (IntrinsicElementalFunction
                            Ibits
                            [(Var 2 from)
                            (IntegerConstant 0 (Integer 4))
                            (IntegerConstant 2 (Integer 4))]
                            0
                            (Integer 4)
                            ()
                        )
                        ()
                    )
                    (If
                        (IntegerCompare
                            (Var 2 res)
                            NotEq
                            (IntegerConstant 2 (Integer 4))
                            (Logical 4)
                            ()
                        )
                        [(ErrorStop
                            ()
                        )]
                        []
                    )
                    (Assignment
                        (Var 2 res8)
                        (IntrinsicElementalFunction
                            Ibits
                            [(Var 2 from8)
                            (IntegerConstant 0 (Integer 4))
                            (IntegerConstant 2 (Integer 4))]
                            0
                            (Integer 8)
                            ()
                        )
                        ()
                    )
                    (If
                        (IntegerCompare
                            (Var 2 res8)
                            NotEq
                            (IntegerConstant 2 (Integer 8))
                            (Logical 4)
                            ()
                        )
                        [(ErrorStop
                            ()
                        )]
                        []
                    )
                    (Assignment
                        (Var 2 from)
                        (IntegerUnaryMinus
                            (IntegerConstant 20 (Integer 4))
                            (Integer 4)
                            (IntegerConstant -20 (Integer 4))
                        )
                        ()
                    )
                    (Assignment
                        (Var 2 from8)
                        (IntegerUnaryMinus
                            (IntegerConstant 20 (Integer 8))
                            (Integer 8)
                            (IntegerConstant -20 (Integer 8))
                        )
                        ()
                    )
                    (Assignment
                        (Var 2 res)
                        (IntrinsicElementalFunction
                            Ibits
                            [(Var 2 from)
                            (IntegerConstant 29 (Integer 4))
                            (IntegerConstant 2 (Integer 4))]
                            0
                            (Integer 4)
                            ()
                        )
                        ()
                    )
                    (If
                        (IntegerCompare
                            (Var 2 res)
                            NotEq
                            (IntegerConstant 3 (Integer 4))
                            (Logical 4)
                            ()
                        )
                        [(ErrorStop
                            ()
                        )]
                        []
                    )
                    (Assignment
                        (Var 2 res8)
                        (IntrinsicElementalFunction
                            Ibits
                            [(Var 2 from8)
                            (IntegerConstant 29 (Integer 4))
                            (IntegerConstant 2 (Integer 4))]
                            0
                            (Integer 8)
                            ()
                        )
                        ()
                    )
                    (If
                        (IntegerCompare
                            (Var 2 res8)
                            NotEq
                            (Cast
                                (IntegerConstant 3 (Integer 4))
                                IntegerToInteger
                                (Integer 8)
                                (IntegerConstant 3 (Integer 8))
                            )
                            (Logical 4)
                            ()
                        )
                        [(ErrorStop
                            ()
                        )]
                        []
                    )
                    (Assignment
                        (Var 2 res)
                        (IntrinsicElementalFunction
                            Ibits
                            [(Var 2 from)
                            (IntegerConstant 2 (Integer 4))
                            (IntegerConstant 2 (Integer 4))]
                            0
                            (Integer 4)
                            ()
                        )
                        ()
                    )
                    (If
                        (IntegerCompare
                            (Var 2 res)
                            NotEq
                            (IntegerConstant 3 (Integer 4))
                            (Logical 4)
                            ()
                        )
                        [(ErrorStop
                            ()
                        )]
                        []
                    )
                    (Assignment
                        (Var 2 res8)
                        (IntrinsicElementalFunction
                            Ibits
                            [(Var 2 from8)
                            (IntegerConstant 2 (Integer 4))
                            (IntegerConstant 2 (Integer 4))]
                            0
                            (Integer 8)
                            ()
                        )
                        ()
                    )
                    (If
                        (IntegerCompare
                            (Var 2 res8)
                            NotEq
                            (Cast
                                (IntegerConstant 3 (Integer 4))
                                IntegerToInteger
                                (Integer 8)
                                (IntegerConstant 3 (Integer 8))
                            )
                            (Logical 4)
                            ()
                        )
                        [(ErrorStop
                            ()
                        )]
                        []
                    )]
                )
        })
    []
)
